packages/wallet-frontend/src/components/dialogs/WithdrawFundsDialog.tsx

```diff
@@ -0,0 +1,148 @@
+              leave="ease-in duration-200"
+              leaveFrom="opacity-100 translate-y-0"
+              leaveTo="opacity-0 translate-y-4"
+            >
+              <Dialog.Panel className="relative w-full max-w-lg space-y-4 overflow-hidden rounded-lg bg-white p-8 shadow-xl">
+                <Dialog.Title
+                  as="h3"
+                  className="text-center text-2xl font-medium text-green-6"
+                >
+                  Withdraw Funds
+                </Dialog.Title>
+                <div className="px-4">
+                  <Form
+                    form={withdrawFundsForm}
+                    onSubmit={async (data) => {
+                      const response = await accountService.withdraw(data)
+
+                      if (!response) {
+                        openDialog(
+                          <ErrorDialog
+                            onClose={closeDialog}
+                            content="Withdrawal failed. Please try again"
+                          />
+                        )
+                        return
+                      }
+
+                      if (response.success) {
+                        router.replace(router.asPath)
+                        closeDialog()
+                      } else {
+                        const { errors, message } = response
+
+                        if (errors) {
+                          getObjectKeys(errors).map((field) =>
+                            withdrawFundsForm.setError(field, {
+                              message: errors[field]
+                            })
+                          )
+                        }
+                        if (message) {
+                          withdrawFundsForm.setError('root', { message })
+                        }
+                      }
+                    }}
+                  >
+                    <Input
+                      disabled={true}
+                      value={account.name}
+                      error={
+                        withdrawFundsForm.formState.errors.accountId?.message
+                      }
+                      label="Account"
+                      readOnly
+                    />
+                    <input
+                      type="hidden"
+                      {...withdrawFundsForm.register('accountId')}
+                      value={account.id}
+                    />
+                    <Input
+                      required
+                      label="Amount"
+                      addOn={getCurrencySymbol(account.assetCode)}
+                      error={
+                        withdrawFundsForm.formState?.errors?.amount?.message
+                      }
+                      {...withdrawFundsForm.register('amount')}
+                    />
+                    <input
+                      type="hidden"
+                      {...withdrawFundsForm.register('assetCode')}
+                      value={account.assetCode}
+                    />
```

Comment:
Is this really needed? This can be grabbed on the backend since the account has all the details about the asset.

Reply:
Yes it is needed, because the whole account is found from the assetCode in the backend, and userID ofcourse.

Reply:
I've just noticed that is happening for the fund dialog as well. We are [passing the account ID](https://github.com/interledger/testnet/blob/ed1e521ec36cc7b18208a3f3b0614609a367fb7f/packages/wallet-frontend/src/lib/api/account.ts#L10) when funding/withdrawing so providing `assetCode` it's redundant. I think we should change this to:
- Only pass the `accountId` and the `amount` - we are performing a query either way to see if the account exists;
- Use the `assetCode` that is retrieved from the database query

https://github.com/interledger/testnet/blob/ed1e521ec36cc7b18208a3f3b0614609a367fb7f/packages/wallet-backend-update/src/account/service.ts#L42-L45
Should be changed to:
```ts
const existingAccount = await Account.query()
  .where('userId', args.userId)
  .where('id', args.accountId)
  .first()
```
or
```ts
const existingAccount = await Account.query().findById(args.accountId)
if(!existingAcount || existingAccount.userId !== args.userId) {
  // throw?
}
```

The backend validation schema should also be changed to accept `accountId`.